### PR TITLE
Upgrade library to use Angular 16 / Ivy

### PR DESCRIPTION
**Pull request by JesseDeBruijne**
- Update from Angular 8 -> Angular 16
- Switch from tslint to eslint and update some linting errors
- Repair unit tests with new Angular version
- Remove unused / old packages

Fixes https://github.com/halloverden/ngx-cookiebot/issues/9

### CODE REVIEW

**Comment by andreaslarssen** [comment]
Nice! Will try and release as soon as I find a minute. Thanks!
PS! Have you tested the build as well?

**Comment by JesseDeBruijne** [comment]
@andreaslarssen How can I test the build? I'm not very familiar with Docker if that's how you build it

**Comment by andreaslarssen** [comment]
> @andreaslarssen How can I test the build? I'm not very familiar with Docker if that's how you build it

We don't use Docker for that, but no worries, I'll test it before releasing it, and it's a major version, so worst case we'll hear about it. Thanks again

**Comment by JesseDeBruijne** [comment]
Hey @andreaslarssen 

I don't mean to sound impatient but I was wondering when you think you have some spare time to check and release these changes. I would love to update my libraries to Angular 16 and I'm sure many other users of this package would too :)

**Comment by andreaslarssen** [comment]
@JesseDeBruijne i know, i'm really sorry that it takes some time. Will try and create an rc version for you to test as soon as i can.

**Comment by andreaslarssen** [comment]
@JesseDeBruijne I changed the target branch so we can create a npm package and test it before merging to master

File: projects/ngx-cookiebot/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "url": "https://github.com/halloverden/ngx-cookiebot/tree/main/projects/ngx-cookiebot"
   },
   "peerDependencies": {
-    "@angular/common": "8.2.14 - 15",
-    "@angular/core": "8.2.14 - 15"
+    "@angular/common": "9 - 16",
+    "@angular/core": "9 - 16"
```
Comment (andreaslarssen) [inline projects/ngx-cookiebot/package.json:34]: If you're not able to test each version from 9-16, I suggest you change both these to  ^16.0.0, and we'll see later if we want to extend version support or release a new major when A17 comes around

**Review by andreaslarssen — CHANGES_REQUESTED** [review]

**Comment by andreaslarssen** [comment]
@JesseDeBruijne `3.0.0-rc.1` has been pushed. Could you install it and verify that it's good?

File: projects/ngx-cookiebot/package.json (hunk 1 above)
Comment (andreaslarssen) [reply projects/ngx-cookiebot/package.json:34]: Fixed it after merging

**Comment by JesseDeBruijne** [comment]
Hey @andreaslarssen 

I have installed the `3.0.0-rc.1` version in an Angular 16 app and it works just fine! The dialog opens as expected, no errors during building or in runtime.

**Pull request merged**